cmake/kokkos_tpls.cmake

```diff
@@ -85,6 +85,11 @@ ENDIF()
 KOKKOS_IMPORT_TPL(ROCM INTERFACE)
 KOKKOS_IMPORT_TPL(LIBQUADMATH)
 
+IF (Kokkos_ENABLE_IMPL_DESUL_ATOMICS AND Kokkos_ENABLE_DESUL_ATOMICS_EXTERNAL)
+  find_package(desul REQUIRED QUIET COMPONENTS atomics)
```

Comment:
Why QUIET?

Reply:
I just kept what we had before. `REQUIRED QUIET` still prints the usual error when the package is not found but doesn't print anything if the package is found.

Reply:
You get 
```
CMake Error at cmake/kokkos_tpls.cmake:89 (find_package):
  By not providing "Finddesul_atomics.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "desul_atomics", but CMake did not find one.

  Could not find a package configuration file provided by "desul_atomics"
  with any of the following names:

    desul_atomicsConfig.cmake
    desul_atomics-config.cmake

  Add the installation prefix of "desul_atomics" to CMAKE_PREFIX_PATH or set
  "desul_atomics_DIR" to a directory containing one of the above files.  If
  "desul_atomics" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  cmake/kokkos_tribits.cmake:244 (INCLUDE)
  CMakeLists.txt:188 (KOKKOS_SETUP_BUILD_ENVIRONMENT)
```
if `desul` wasn't found.

I don't have a strong opinion on using `QUIET` or not but I think it makes sense conceptionally since we print anyway if a bundled or external package is used.
(At the moment, we don't print anything anyway without `QUIET`.)

Reply:
Drop it please